modules/flowable-cmmn-engine-configurator/src/main/java/org/flowable/cmmn/engine/configurator/impl/cmmn/DefaultCaseInstanceService.java

```diff
@@ -109,4 +114,20 @@ public void deleteCaseInstance(String caseInstanceId) {
         cmmnEngineConfiguration.getCmmnRuntimeService().terminateCaseInstance(caseInstanceId);
     }
 
+    @Override
+    public Set<String> findChildCaseIdsForExecutionIds(Collection<String> executionIds) {
+        CmmnRuntimeService cmmnRuntimeService = cmmnEngineConfiguration.getCmmnRuntimeService();
+        CaseInstanceQuery caseInstanceQuery = cmmnRuntimeService.createCaseInstanceQuery()
+                .caseInstanceCallbackType(CallbackTypes.EXECUTION_CHILD_CASE)
+                .or();
+        for (String executionId : executionIds) {
+            caseInstanceQuery.caseInstanceCallbackId(executionId);
```

Comment:
This is not going to work properly. The query will execute only with the last executionId, it is the same as without the or. The reason for that is that the query can have only one `caseInstanceCallbackId`. If you really need to query by all execution ids then add a `caseInstanceCallbackIds` to the query. In order to verify that this is not working you can add a test where you have 2 parallel case tasks and it won't pass.

Reply:
Thanks! I will fix it.

Reply:
fixed